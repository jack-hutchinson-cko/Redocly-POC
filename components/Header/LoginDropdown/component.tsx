import React from 'react';
import {
	StyledApplyLink,
	StyledBottom,
	StyledButtonContainer,
	StyledContainer,
	StyledTop,
	StyledLabel,
	StyledLink,
	StyledText,
	StyledTextBottom,
} from './loginDropdown.styles';
import { ArrowRight, TheHub, TestAccount } from '../../svg';
import ButtonLink from '../../ButtonLink';

const LoginDropdown = (): JSX.Element => (
	<StyledContainer>
		<StyledTop>
			<StyledLabel>
				<TheHub />
				The Hub
			</StyledLabel>
			<StyledText>Monitor transactions, business performance and customer trends.</StyledText>
			<StyledButtonContainer>
				<ButtonLink to="https://hub.checkout.com/login" hierarchy="secondary">
					Log In
				</ButtonLink>
				<StyledText>or</StyledText>
				<StyledApplyLink href="https://www.checkout.com/get-test-account">
					Apply for an account
					<ArrowRight />
				</StyledApplyLink>
			</StyledButtonContainer>
		</StyledTop>
		<StyledBottom>
			<StyledLabel>
				<TestAccount />
				Test account
			</StyledLabel>
			<StyledTextBottom>
				<StyledLink href="https://sandbox.checkout.com/login">Log in</StyledLink>
				or
				<StyledLink href="https://www.checkout.com/get-test-account">apply for a test account</StyledLink>
			</StyledTextBottom>
		</StyledBottom>
	</StyledContainer>
);

export default LoginDropdown;
