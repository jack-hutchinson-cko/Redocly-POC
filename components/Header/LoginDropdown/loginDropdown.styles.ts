import styled from 'styled-components';
import { colours } from '../../css/tokens';

export const StyledContainer = styled.div`
	background: white;
	border: 1px solid ${colours.S05};
	box-shadow: 0px 2px 5px rgba(12, 17, 66, 0.15);
	color: ${colours.S1000};
	display: flex;
	flex-direction: column;
	font-family: 'Graphik LCG Web';
	position: relative;
	width: 434px;
	position: absolute;
	top: 64px;
	right: -140px;
	z-index: 1000;
`;

export const StyledSVG = styled.div`
	display: flex;
	margin: auto;
	transform: ${(props: { isOpen: boolean }) => (props.isOpen ? 'rotate(180deg)' : 'none')};
`;

export const StyledLabel = styled.div`
	display: flex;
	align-items: center;
	gap: 12px;
	weight: 500;
`;

export const StyledTop = styled.div`
	padding: 40px 32px;
`;

export const StyledBottom = styled.div`
	background: ${colours.S05};
	padding: 40px 32px;
`;

export const StyledText = styled.p`
	display: flex;
	gap: 8px;
`;

export const StyledTextBottom = styled.p`
	display: flex;
	gap: 8px;
	margin-left: 30px;
`;

export const StyledApplyLink = styled.a`
	align-items: center;
	display: flex;
	color: ${colours.S1000};
	font-size: 14px;
	font-weight: 500;
	gap: 8px;
	line-height: 24px;
	letter-spacing: 0em;
	text-align: left;
	margin: auto 0;
	padding: 0px;
	text-decoration: none;

	:hover {
		text-decoration: none;
	}
`;

export const StyledArrow = styled.div`
	display: inline;
	padding-top: 2px;
`;

export const StyledLink = styled.a`
	color: ${colours.S1000};
	text-decoration: underline;
`;

export const StyledButtonContainer = styled.div`
	display: flex;
	gap: 16px;
	margin: auto;
`;
