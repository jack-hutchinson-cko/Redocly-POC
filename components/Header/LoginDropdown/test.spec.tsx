import { shallow } from 'enzyme';
import React from 'react';
import LoginDropdown from './component';

describe('LoginDropdown component', () => {
	test('component matches snapshot', () => {
		expect.assertions(1);

		const wrapper = shallow(<LoginDropdown />);

		expect(wrapper).toMatchSnapshot();
	});
});
