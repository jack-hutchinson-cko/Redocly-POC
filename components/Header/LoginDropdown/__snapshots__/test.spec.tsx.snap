// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginDropdown component component matches snapshot 1`] = `
<styled.div>
  <styled.div>
    <styled.div>
      <TheHub />
      The Hub
    </styled.div>
    <styled.p>
      Monitor transactions, business performance and customer trends.
    </styled.p>
    <styled.div>
      <ButtonLink
        hierarchy="secondary"
        to="https://hub.checkout.com/login"
      >
        Log In
      </ButtonLink>
      <styled.p>
        or
      </styled.p>
      <styled.a
        href="https://www.checkout.com/get-test-account"
      >
        Apply for an account
        <ArrowRight />
      </styled.a>
    </styled.div>
  </styled.div>
  <styled.div>
    <styled.div>
      <TestAccount />
      Test account
    </styled.div>
    <styled.p>
      <styled.a
        href="https://sandbox.checkout.com/login"
      >
        Log in
      </styled.a>
      or
      <styled.a
        href="https://www.checkout.com/get-test-account"
      >
        apply for a test account
      </styled.a>
    </styled.p>
  </styled.div>
</styled.div>
`;
